### PR TITLE
Add Documentation for Track 2 Management SDK

### PR DESCRIPTION
**Pull request by nickzhums**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,516 +46,29 @@ If a package you need isn't available please open an issue and let us know.
...
-autorest.WithSendDecorators(ctx, []autorest.SendDecorator{
-	autorest.DoRetryForStatusCodesWithCap(client.RetryAttempts,
-		client.RetryDuration, time.Duration(0),
-		autorest.StatusCodesForRetry...)})
-client.List(ctx)
-```
-
-This will replace the default slice of `SendDecorators` with the provided slice.
-
-The `PollingDelay` and `PollingDuration` values are used exclusively by [WaitForCompletionRef()](https://godoc.org/github.com/Azure/go-autorest/autorest/azure#Future.WaitForCompletionRef) when blocking on an async call until it completes.
```
Comment (nickzhums) [inline README.md:525]: this content was too long. Moved it into another file and provided a link to it

File: documentation/new-version-quickstart.md [hunk 2]
```diff
@@ -0,0 +1,692 @@
+
```
Comment (nickzhums) [inline documentation/new-version-quickstart.md:1]: this is the Track 2 quickstart and will eventually be moved into Microsoft Docs. Providing it here for now so customer can access it

File: documentation/previous-versions-quickstart.md [hunk 3]
```diff
@@ -0,0 +1,531 @@
+# Azure SDK for Go - Previous Versions
```
Comment (nickzhums) [inline documentation/previous-versions-quickstart.md:1]: this is the document for track 1 and previously on readme page (it is very long, so moving to here to provide clear distinction between track 1 and track 2)

File: documentation/new-version-quickstart.md [hunk 4]
```diff
@@ -0,0 +1,692 @@
...
+***Creating a Resource Group***
+```go
+func createResourceGroup(connection *armcore.Connection) error {
+	rgClient := armresources.NewResourceGroupsClient(connection, subscriptionId)
+
+	param := armresources.ResourceGroup{
+		Location: to.StringPtr(location),
+	}
+
+	resp, err := rgClient.CreateOrUpdate(ctx, resourceGroupName, param, nil)
```
Comment (ArcturusZhang) [inline documentation/new-version-quickstart.md:416]: Just return `rgClient.CreateOrUpdate(ctx, resourceGroupName, param, nil)` should be sufficient, just like the example above. We need to keep things consistent. The return type need adjusting accordingly
```suggestion
	return rgClient.CreateOrUpdate(ctx, resourceGroupName, param, nil)
```

File: documentation/new-version-quickstart.md [hunk 5]
```diff
@@ -0,0 +1,692 @@
...
+		},
+	}
+	poller, err := vnetClient.BeginCreateOrUpdate(ctx, resourceGroupName, vnetName, param, nil)
+	if err != nil {
+		return err
+	}
+	resp, err := poller.PollUntilDone(ctx, interval)
+	if err != nil {
+		return err
+	}
```
Comment (ArcturusZhang) [inline documentation/new-version-quickstart.md:455]:
```suggestion
	return poller.PollUntilDone(ctx, interval)
```

File: documentation/new-version-quickstart.md [hunk 6]
```diff
@@ -0,0 +1,692 @@
...
+
+	resourceGroupID = *resp.ResourceGroup.ID
+	fmt.Printf("Resource Group '%s' created: \n%s\n", resourceGroupID, string(b))
+	return nil
+}
+```
+
+***Creating a Virtual Network***
+```go
+func createVirtualNetwork(connection *armcore.Connection) error {
```
Comment (ArcturusZhang) [inline documentation/new-version-quickstart.md:433]:
```suggestion
func createVirtualNetwork(connection *armcore.Connection) (armnetwork.VirtualNetworkResponse, error) {
```

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]
I doubt if it is necessary to break up a full sample program into pieces and put it in the quick-start. But those pieces need to be consistent with each other and have the same style.

I propose that we could make the new-version-quickstart brief and add a reference to the full sample program for reference.

File: README.md [hunk 7]
```diff
@@ -3,42 +3,34 @@
...
-It officially supports the last two major releases of Go.  Older versions of
-Go will be kept running in CI until they no longer work due to changes in any
-of the SDK's external dependencies.  The CHANGELOG will be updated when a
-version of Go is removed from CI.
+This repository is for active development of the Azure SDK for Go. For consumers of the SDK you can follow the links below to visit the documentation you are interested in
+* [Overview of Azure SDK for Go](https://docs.microsoft.com/azure/developer/go/)
+* [SDK Reference](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go)
+* [Code Samples for Azure Go SDK](https://github.com/azure-samples/azure-sdk-for-go-samples)
+* [Azure REST API Docs](https://docs.microsoft.com/rest/api/)
+* [General Azure Docs](https://docs.microsoft.com/azure)
```
[the remark this answers is not in the document]
Comment (nickzhums) [reply README.md:11]: REST API docs and general Azure docs can be applied to both. I think we can add a link for pkg.go.dev docs. 

For samples repo, we expect to update them soon

**Comment by nickzhums** [comment]
> I doubt if it is necessary to break up a full sample program into pieces and put it in the quick-start. But those pieces need to be consistent with each other and have the same style.
> 
> I propose that we could make the new-version-quickstart brief and add a reference to the full sample program for reference.

Makes sense. For full end to end sample, can we add them to the samples repo so I can provide links to them?

File: documentation/new-version-quickstart.md [hunk 8]
```diff
@@ -0,0 +1,692 @@
...
+			IncludeBody: true,
+		},
+	})
+
+	defer cleanup(conn)
+
+	if err := createResourceGroup(ctx, conn); err != nil {
+		panic(err)
+	}
+})
```
[the remark this answers is not in the document]
Comment (nickzhums) [reply documentation/new-version-quickstart.md:195]: i think the feedback we got is that they want to see a complete sample whenever possible. We should at least show one complete example here and maybe move the rest into samples repo

File: documentation/previous-versions-quickstart.md [hunk 9]
```diff
@@ -0,0 +1,531 @@
...
+
+- SDK docs are at [godoc.org](https://godoc.org/github.com/Azure/azure-sdk-for-go/).
+- SDK samples are at [Azure-Samples/azure-sdk-for-go-samples](https://github.com/Azure-Samples/azure-sdk-for-go-samples).
+- SDK notifications are published via the [Azure update feed](https://azure.microsoft.com/updates/).
+- Azure API docs are at [docs.microsoft.com/rest/api](https://docs.microsoft.com/rest/api/).
+- General Azure docs are at [docs.microsoft.com/azure](https://docs.microsoft.com/azure).
+
+## Reporting security issues and security bugs
+
+Security issues and bugs should be reported privately, via email, to the Microsoft Security Response Center (MSRC) <secure@microsoft.com>. You should receive a response within 24 hours. If for some reason you do not, please follow up via email to ensure we recei
```
Comment (lirenhe) [inline documentation/previous-versions-quickstart.md:531]: Seems the last sentence is not completed.

Comment (nickzhums) [reply documentation/previous-versions-quickstart.md]: thanks, added them back

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**